### ISSUE TITLE
xrdp_keyboard.ini ignores parameters

### ISSUE THREAD

**Issue by endofreal** [neutrinolabs/xrdp#1660]
Hi!

I try to configure default keyboard layouts.

I try to change values in default section of xrdp_keyboard.ini and it ignores next changes:
```
[default]
keyboard_type=0
keyboard_subtype=0
variant=,dos  # ignored
model=pc105  # ignored
rdp_layouts=default_rdp_layouts
layouts_map=rdp_layouts_map_ru
options=grp:alt_shift_toggle  # ignored

[rdp_layouts_map_ru]
rdp_layout_us=us,ru
rdp_layout_ru=us,ru
```

Output of `setxkbmap -query`:
```
# Connect with RU layout
rules:      base
model:      pc104
layout:     us,ru

# Connect with US layout
rules:      base
model:      pc104
layout:     us,ru
```

What I do wrong?

Environment:

- OS: ubuntu:20.04
- xrdp: 0.9.13.1 (from source)
- xorgxrdp: from source devel branch

**Comment by endofreal** [comment]
> I think it likely the settings you're putting in `xrdp.ini` are being overridden by the system settings when the desktop starts.
> 
> Try playing with `localectl` and/or whatever facilities your window manager gives you for setting the language.

I don't think so.

I using IceWM as my windows manager and pcmanfm-qt as, it does not have any keyboard settings.

Besides keyboard settings will be reset if do reconnect (windows manager in this case do not restart isn't it?).
Case:
```
echo icewm-session > ~/.xsession
# connect
setxkbmap -query
# rules:      base
# model:      pc104
# layout:     us
setxkbmap -model pc105 -layout us,ru -variant dvorak, -option grp:alt_shift_toggle
setxkbmap -query
# rules:      base
# model:      pc105
# layout:     us,ru
# variant:    dvorak,
# options:    grp:alt_shift_toggle
# reconnect
setxkbmap -query
# rules:      base
# model:      pc104
# layout:     us
```

I also tried to use `localectl --no-convert set-x11-keymap us,ru pc105 "dvorak," grp:alt_shift_toggle` it does not have any effect.
Tried to add next section in `/etc/X11/xorg.conf`
```
Section "InputClass"
        Identifier "system-keyboard"
        MatchIsKeyboard "on"
        Option "XkbLayout" "us,ru"
        Option "XkbModel" "pc105"
        Option "XkbVariant" "dvorak,"
        Option "XkbOptions" "grp:alt_shift_toggle"
EndSection
```
And there is no result.

May you need some other information? 

Now as workaround I use `(sleep 5s; setxkbmap -model pc105 -layout us,ru -variant dvorak, -option grp:alt_shift_toggle)&` in `reconnectwm.sh` script.

**Comment by endofreal** [comment]
Any ideas how to resolve it?

**Comment by matt335672** [comment]
I've not gotten on to this one yet. At the moment I haven't I'm afraid, but I'll have a look when I get a chance.

**Comment by nikimagic** [comment]
chmod 777  xrdp_keyboard.ini

**Comment by ValentinChirikov** [comment]
Hello everyone ! Has the issue been resolved ? Looks like i am experiencing the same problem :(

**Comment by nikimagic** [comment]
> Всем привет ! Проблема решена? Похоже, у меня такая же проблема :(

which ?  Какая?

**Comment by ValentinChirikov** [comment]
> > Всем привет ! Проблема решена? Похоже, у меня такая же проблема :(
> 
> which ? Какая?

The keyboard layout not being switched, even i specify settings in xrdp_keyboard.ini
for example:
[default]
keyboard_type=0
keyboard_subtype=0
options=grp:win_space_toggle

[default_layouts_map]
rdp_layout_none=us,ru
rdp_layout_us=us,ru
rdp_layout_us_dvorak=dvorak,ru
rdp_layout_ru=ru,us

full config attached
[xrdp_keyboard.zip](https://github.com/neutrinolabs/xrdp/files/10840999/xrdp_keyboard.zip)

**Comment by nikimagic** [comment]
> > > Всем привет ! Проблема решена? Похоже, у меня такая же проблема :(
> > 
> > 
> > which ? Какая?
> 
> The keyboard layout not being switched, even i specify settings in xrdp_keyboard.ini for example: [default] keyboard_type=0 keyboard_subtype=0 options=grp:win_space_toggle
> 
> [default_layouts_map] rdp_layout_none=us,ru rdp_layout_us=us,ru rdp_layout_us_dvorak=dvorak,ru rdp_layout_ru=ru,us
> 
> full config attached [xrdp_keyboard.zip](https://github.com/neutrinolabs/xrdp/files/10840999/xrdp_keyboard.zip)

Да, во всех кубунтах  приходится русскую раскладку добавлять вручную.  в  22.04  очень хорошо починили, проблема  только с  раскладкой осталась. Поэтому  открывайте файл редактором, добавляйте  нужные  строки, и сохраняйте. все  под  sudo
Эта история лет 15  уже, как я вижу по всяким  форумам
